lib/rules/no-restricted-imports.js

```diff
@@ -248,26 +219,38 @@ module.exports = {
          */
         function checkNode(node) {
             const importSource = node.source.value.trim();
-            const importNames = node.specifiers ? node.specifiers.reduce((set, specifier) => {
+            const importNames = node.specifiers ? node.specifiers.reduce((map, specifier) => {
+                const specifierData = { loc: specifier.loc };
+
                 if (specifier.type === "ImportDefaultSpecifier") {
-                    set.add("default");
+                    map.set("default", [specifierData]);
                 } else if (specifier.type === "ImportNamespaceSpecifier") {
-                    set.add("*");
+                    map.set("*", specifierData);
                 } else if (specifier.imported) {
-                    set.add(specifier.imported.name);
+                    if (map.has(specifier.imported.name)) {
+                        const specifiers = map.get(specifier.imported.name);
+
+                        specifiers.push(specifierData);
+                        map.set(specifier.imported.name, specifiers);
+                    } else {
+                        map.set(specifier.imported.name, [specifierData]);
+                    }
                 } else if (specifier.local) {
-                    set.add(specifier.local.name);
+                    if (map.has(specifier.local.name)) {
+                        const specifiers = map.get(specifier.local.name);
+
+                        specifiers.push(specifierData);
+                        map.set(specifier.local.name, specifiers);
+                    } else {
+                        map.set(specifier.local.name, [specifierData]);
+                    }
                 }
```

Comment:
This works fine! But wondering would it be easier to follow if we would just get the `name` in `if-else-if`s and then deal with the map after the chain? Then we wouldn't need to duplicate code for `imported` and `local` and think about why is `default` initialized as an array while `*` isn't. 

Code that checks `*` in `checkRestrictedPathAndReport` could take just `[0]`.

Thoughts?

Reply:
In this way it looks much more clearer and no duplication of the code. Awesome! Thank you!